x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -364,4 +370,19 @@ protected XPackLicenseState getLicenseState() {
                 CACHE_PREWARMING_THREAD_POOL_SETTING
             ) };
     }
+
+    public static final class CacheServiceSupplier implements Supplier<CacheService> {
```

Comment:
Best solution I could think of for passing the cache to the transport action (which is instantiated on master and data nodes but only ever handles the fan-out request on data nodes) on master- and data-nodes without instantiating the cache on master as well.